inputremapper/injection/macros/macro.py

```diff
@@ -676,17 +676,19 @@ async def listener(event):
 
             resolved_timeout = _resolve(timeout, allowed_types=[int, float, None])
             await asyncio.wait(
-                [listener_done.wait(), self._trigger_release_event.wait()],
+                [asyncio.Task(listener_done.wait()), asyncio.Task(self._trigger_release_event.wait())],
                 timeout=resolved_timeout / 1000 if resolved_timeout else None,
                 return_when=asyncio.FIRST_COMPLETED,
             )
 
             self.context.listeners.remove(listener)
 
             if not listener_done.is_set() and self._trigger_release_event.is_set():
-                await then.run(handler)  # was trigger release
+                if then:
+                    await then.run(handler)  # was trigger release
             else:
-                await else_.run(handler)
+                if else_:
+                    await else_.run(handler)
```

Comment:
```py
            if not listener_done.is_set() and self._trigger_release_event.is_set() and then:
                await then.run(handler)  # was trigger release
            elif else_:
                await else_.run(handler)
```

Reply:
however, this exceeds the maximum 88 chars of black, you'll have to extract `self._trigger_release_event.is_set()` and `not listener_done.is_set()` into variables with sensible names to make that line shorter.

Reply:
ah wait, this is probably wrong. It would always run else_ if then is not set.